### PR TITLE
[TE][notification] Jira Alerting improvements

### PR DESCRIPTION
**Pull request by akshayrai**
Changes:
* Try to auto-populate required jira fields based on allowed values for that field.
* Added alert snapshot - Whenever we report a single anomaly, ThirdEye will attach the alert timeseries snapshot to the jira ticket.
* Added ability to configure the jira components field
* Fixed minor typo in the jira template

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/notification/commons/ThirdEyeJiraClient.java [hunk 1]
```diff
@@ -135,33 +161,82 @@ public void reopenIssue(Issue issue) {
...
+        .withProjectKeys(jiraEntity.getJiraProject())
+        .withIssueTypeIds(jiraEntity.getJiraIssueTypeId())
+        .withExpandedIssueTypesFields()
+        .build()).claim();
+  }
+
+  private Map<String, CimFieldInfo> getIssueRequiredCreateFields(JiraEntity jiraEntity) {
+    Map<String, CimFieldInfo> requiredCreateFields = new HashMap<>();
+    CimProject project = getProjectMetadata(jiraEntity).iterator().next();
+    CimIssueType issueType = project.getIssueTypes().iterator().next();
```
Comment (jihaozh) [inline thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/notification/commons/ThirdEyeJiraClient.java]: should we check `iterator().hasNext()` first?

Comment (akshayrai) [reply thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/notification/commons/ThirdEyeJiraClient.java]: Added the check

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/notification/formatter/channels/JiraContentFormatter.java [hunk 2]
```diff
@@ -48,6 +47,8 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import static org.apache.pinot.thirdeye.notification.commons.ThirdEyeJiraClient.*;
```
Comment (vincentchenjl) [inline thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/notification/formatter/channels/JiraContentFormatter.java]: Can we include the classes explicitly? If there are not too many of them, it is better to include explicitly.

Comment (akshayrai) [reply thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/notification/formatter/channels/JiraContentFormatter.java]: I double checked. There is actually more than 5 constants we depend on.

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/notification/commons/ThirdEyeJiraClient.java [hunk 3]
```diff
@@ -62,12 +78,11 @@ private JiraRestClient createJiraRestClient(JiraConfiguration jiraAdminConfig) {
...
 
   private String buildQueryOnCreatedBy(long lookBackMillis) {
-    long lookBackDays = TimeUnit.MILLISECONDS.toDays(lookBackMillis);
     String createdByQuery = "created";
-    if (lookBackDays < 0) {
+    if (lookBackMillis < 0) {
       createdByQuery += "<= -0m";
     } else {
-      createdByQuery += ">= -" + lookBackDays + "d";
+      createdByQuery += ">= -" + TimeUnit.MILLISECONDS.toDays(lookBackMillis) + "d";
```
[the remark this answers is not in the document]
Comment (akshayrai) [reply thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/notification/commons/ThirdEyeJiraClient.java]: There is no change in logic here. `TimeUnit.MILLISECONDS.toDays` should convert millisec to days.

**Pull request merged**